libs/execution_context/ebpf_program.c

```diff
@@ -2400,7 +2400,7 @@ _ebpf_program_test_run_work_item(_In_ cxplat_preemptible_work_item_t* work_item,
     void* program_context = NULL;
     bool supports_context_header;
 
-    result = ebpf_set_current_thread_affinity((uintptr_t)1 << options->cpu, &old_thread_affinity);
+    result = ebpf_set_current_thread_cpu_affinity(options->cpu, &old_thread_affinity);
```

Comment:
I highly recommend not using NT kernel processor indices in code reachable from user mode.

This is a conundrum, though, because I bet Linux uses integer processor IDs that are consistent between user mode and kernel mode.

On Windows, the only unambiguous way to specify a processor from user mode is via the PROCESSOR_NUMBER struct, i.e., by specifying a group and group-relative integer.

Reply:
We have no control over the contract with user mode. What is the problem with using processor index from user mode?

Reply:
Windows does not provide a notion of an integer processor index to user mode applications. If you hover over processors in task manager, for example, they will tell you a node-relative CPU index that they cook up themselves.

All user mode APIs, and the only standardized/canonical representation of a CPU identifier is the PROCESSOR_NUMBER. There are no user mode APIs to convert between PROCESSOR_NUMBER and a standard, unambiguous ULONG index, in other words.

Reply:
Well, for now I think we keep compat with the libbpf API and if someone needs this we can add an ebpf-for-windows only API for this.